act/retrievals/sirs_calculations.py

```diff
@@ -0,0 +1,187 @@
+
+""" Calculating SIRS Radiation Estimates """
+
+import numpy as np
+from astral import sun
+from scipy.constants import Stefan_Boltzmann
+from datetime import datetime, timedelta
+import astral
+
+def numpy_to_datetime(date_time64_val):
+    '''
+    Converts a np datetime64 value to a standard python datetime value.
+
+    ...
+
+    Parameters
+    ----------
+    date_time64_val : numpy.datetime64 or list of numpy.datetime64
+                      or numpy array of numpy.datetime64
+        A numpy.datetime64 object value
+
+    Returns
+    --------
+    time : datetime object or numpy array of datetime object
+        Datetime object converted from numpy dattime64 object.
+
+    '''
+
+    py_datetime_val = (
+        date_time64_val -
+        np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')
+
+    return py_datetime_val
+
+
+def calculate_sirs_variables(sirs_obj, met_obj, sirs_time='time', met_time='time', lat='lat', lon='lon', 
```

Comment:
Would it make sense to split this out into different functions for each calculated variable?

Reply:
We also probably want to rename this to a more general name like radiation.py or something like that.  Might help with non-ARM lingo knowing users :)

Reply:
I don't know - some of the calculations are tied together (for example, you need the derived down short hemispheric value to get the irradiance difference and irradiance ratio), and they are all pretty closely related, so my first thought would be that it makes more sense to keep them all together. What do you think?

And, sounds good - I will change the name to something more generic.